### PR TITLE
Add task3

### PR DESCRIPTION
**Pull request by iharyachnik**

### CODE REVIEW

File: server/utils/streams.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    })
+    .option('path', {
+      abbr: 'p',
+      help: 'Directory path'
+    })
+    .parse();
+
+  if (action) {
+    switch (action) {
+      case 'transform':
```
Comment (DraXu3) [inline server/utils/streams.js]: Move case values to constants

File: server/utils/streams.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+      case 'bundle-css':
+        bundle(path);
+        break;
+    }
+  } else {
+    console.log('Usage: node streams.js [action] [options]');
+  }
+}
+
+function help(param) {
```
Comment (DraXu3) [inline server/utils/streams.js:58]: Please export all supplementary functions

File: server/utils/streams.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+  });
+
+  createPipe(process.stdin, process.stdout, converter);
+}
+
+function bundle(dirPath) {
+  if (!dirPath) return help('path');
+
+  const converter = () => through(function (chunk, enc, cb) {
+    console.log(chunk.toString());
```
Comment (DraXu3) [inline server/utils/streams.js]: Please remove this redundant console log

File: server/utils/streams.js [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    let first = true;
+
+    return () => {
+      if (first) {
+        first = false;
+        return null;
+      } else {
+        return flags;
+      }
+    }
```
Comment (DraXu3) [inline server/utils/streams.js]: Please add missing semicolon

File: server/utils/streams.js [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+  if (!dirPath) return help('path');
+
+  const converter = () => through(function (chunk, enc, cb) {
+    console.log(chunk.toString());
+    this.push(`${chunk.toString()}\n`);
+
+    cb();
+  });
+
+  const flags = (() => {
```
Comment (DraXu3) [inline server/utils/streams.js]: Please move this logic to a separate named function

File: server/utils/streams.js [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+        return null;
+      } else {
+        return flags;
+      }
+    }
+  })();
+
+  function next(source, converter, flags) {
+    return new Promise((res, rej) => {
+      const dest = fs.createWriteStream('bundle.css', flags);
```
Comment (DraXu3) [inline server/utils/streams.js]: Please move this to a constant

File: server/utils/streams.js [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+    });
+  }
+
+  readDir(dirPath)
+    .then(files => {
+      return files
+        .filter(file => path.extname(file) === '.css')
+        .map(file => getFilePath(dirPath, file))
+        .map(file => fs.createReadStream(file))
+        .concat(request('https://www.epam.com/etc/clientlibs/foundation/main.min.fc69c13add6eae57cd247a91c7e26a15.css'));
```
Comment (DraXu3) [inline server/utils/streams.js]: Please move this url to a constant

File: server/utils/streams.js [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+      streams
+        .reduce((p, file) => p.then(() => next(file, converter(), flags())), Promise.resolve());
+    })
+}
+
+function getJsonConverter() {
+  let first = true;
+
+  return through(function (chunk, enc, cb) {
+    first && this.push('[');
```
Comment (DraXu3) [inline server/utils/streams.js]: Please refactor these condition-like expressions

File: server/utils/streams.js (hunk 2 above)
Comment (iharyachnik) [reply server/utils/streams.js:58]: Also fixed, but this string was not changed :)

**Pull request merged**